lib/dmv.rb

```diff
@@ -9,8 +10,48 @@ def add_facility(facility)
   end
 
   def facilities_offering_service(service)
-    @facilities.find do |facility|
+    @facilities.find_all do |facility|
       facility.services.include?(service)
     end
   end
-end
+
+  def format_address(dmv_site)
+    if dmv_site.has_key?(:address_li)
+      dmv_site[:address_li] + " " + dmv_site[:address__1].to_s + " " + dmv_site[:city] + " " + dmv_site[:state] + " " + dmv_site[:zip]
+    elsif dmv_site.has_key?(:street_address_line_1)
+      dmv_site[:street_address_line_1] + " " + dmv_site[:city] + " " + dmv_site[:state] + " " + dmv_site[:zip_code]
+    elsif dmv_site.has_key?(:address1)
+      dmv_site[:address1] + " " + dmv_site[:city] + " " + dmv_site[:state] + " " + dmv_site[:zipcode]
+    end
+  end
+
+  def format_name(dmv_site)
+    if dmv_site.has_key?(:dmv_office)
+      dmv_site[:dmv_office]
+    elsif dmv_site.has_key?(:office_name)
+      dmv_site[:office_name]
+    elsif dmv_site.has_key?(:name)
+      dmv_site[:name]
+    end
+  end
+
+  def format_phone(dmv_site)
+    if dmv_site.has_key?(:phone)
+      dmv_site[:phone]
+    elsif dmv_site.has_key?(:public_phone_number)
+      dmv_site[:public_phone_number]
+    end
+  end
+
+  def create_facilities(dmv_location_data)
+    dmv_location_data.map do |dmv_site|
+      dmv_facility_details = {
+        name: format_name(dmv_site),
+        address: format_address(dmv_site),
+        phone: format_phone(dmv_site)
+      }
+
+      Facility.new(dmv_facility_details)
+    end
+  end
```

Comment:
Yes, I do like how you have separated out all of these concerns into different helper methods. 
In terms of organization, we usually have the Integration methods up top (create_facilities in this case) and the helper methods down below, that way we can quickly see the meat of the logic and then if we need to dig into the helper methods, we can do that. Otherwise great job. 
Another thought is you could use Boolean Operators instead of if statements so something like this for the `format_phone` method:
`dmv_site[:phone] || dmv_site[:public_phone_number]`
This would look at the first expression and if it has a value, would return that value, if no value (nil), then it would look to the second expression and if there's a value there, would return that value. If no values are found on either side, this would return false.